src/config/authd-config.c

```diff
@@ -192,4 +216,99 @@ short eval_bool(const char *str) {
         return OS_INVALID;
     }
 }
-#endif
\ No newline at end of file
+
+int get_time_interval(char *source, time_t *interval) {
+    char *endptr;
+    *interval = strtoul(source, &endptr, 0);
+
+    if ((!*interval && endptr == source) || *interval < 0) {
+        return OS_INVALID;
+    }
+
+    switch (*endptr) {
+    case 'd':
+        *interval *= 86400;
+        break;
+    case 'h':
+        *interval *= 3600;
+        break;
+    case 'm':
+        *interval *= 60;
+        break;
+    case 's':
+    case '\0':
+        break;
+    default:
+        return OS_INVALID;
+    }
+
+    return 0;
+}
+
+int w_read_force_config(XML_NODE node, authd_config_t *config) {
+    /* XML Definitions */
+    static const char *xml_enabled = "enabled";
+    static const char *xml_key_mismatch = "key_mismatch";
+    static const char *xml_disconnected_time = "disconnected_time";
+    static const char *xml_after_registration_time = "after_registration_time";
+
+    for (int i = 0; node[i]; i++) {
+        // enabled
+        if (!strcmp(node[i]->element, xml_enabled)) {
+            short b = eval_bool(node[i]->content);
+
+            if (b < 0) {
+                merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                return OS_INVALID;
+            }
+
+            config->force_options.enabled = (b ? true : false);
+        }
+        // key_mismatch
+        else if (!strcmp(node[i]->element, xml_key_mismatch)) {
+            short b = eval_bool(node[i]->content);
+
+            if (b < 0) {
+                merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                return OS_INVALID;
+            }
+
+            config->force_options.key_mismatch = (b ? true : false);
+        }
+        // disconnected_time
+        else if (!strcmp(node[i]->element, xml_disconnected_time)) {
+            if (node[i]->attributes && node[i]->attributes[0] && !strcmp(node[i]->attributes[0], xml_enabled)) {
+                if (node[i]->values && node[i]->values[0]) {
+
+                    short b = eval_bool(node[i]->values[0]);
+
+                    if (b < 0) {
+                        merror(XML_VALUEERR, node[i]->attributes[0], node[i]->values[0]);
+                        return OS_INVALID;
+                    } else if (b > 0) {
+                        config->force_options.disconnected_time_enabled = true;
+                        if (!strcmp(node[i]->element, xml_disconnected_time)) {
```

Comment:
This `if (!strcmp(node[i]->element, xml_disconnected_time))` is the same than the line 279, it isn't necessary again.